azext_edge/tests/edge/init/test_base_unit.py

```diff
@@ -726,3 +728,115 @@ def test_verify_custom_locations_enabled(mocker, role_bindings):
+                subscription_id=ZEROED_SUB,
+                cluster_name="cluster1",
+                resource_group_name="rg1",
+            ),
+        },
+        {  # fail config indicates diff cluster
+            "failure": "cluster name",
+            "config_map": {
+                "apiVersion": "v1",
+                "data": {
+                    "AZURE_RESOURCE_NAME": "cluster2",
+                    "AZURE_RESOURCE_GROUP": "rg1",
+                    "AZURE_SUBSCRIPTION_ID": ZEROED_SUB,
+                },
+                "metadata": {
+                    "name": "azure-clusterconfig",
+                    "namespace": "azure-arc",
+                },
+            },
+            "connected_cluster": ConnectedCluster(
+                subscription_id=ZEROED_SUB,
+                cluster_name="cluster1",
+                resource_group_name="rg1",
+            ),
+        },
+        {  # fail config indicates diff rg
+            "failure": "resource group",
+            "config_map": {
+                "apiVersion": "v1",
+                "data": {
+                    "AZURE_RESOURCE_NAME": "cluster1",
+                    "AZURE_RESOURCE_GROUP": "rg2",
+                    "AZURE_SUBSCRIPTION_ID": ZEROED_SUB,
+                },
+                "metadata": {
+                    "name": "azure-clusterconfig",
+                    "namespace": "azure-arc",
+                },
+            },
+            "connected_cluster": ConnectedCluster(
+                subscription_id=ZEROED_SUB,
+                cluster_name="cluster1",
+                resource_group_name="rg1",
+            ),
+        },
+        {  # fail config indicates diff sub
+            "failure": "subscription Id",
+            "config_map": {
+                "apiVersion": "v1",
+                "data": {
+                    "AZURE_RESOURCE_NAME": "cluster1",
+                    "AZURE_RESOURCE_GROUP": "rg1",
+                    "AZURE_SUBSCRIPTION_ID": "8757c60a-a398-4c09-adaf-be328caf42d4",
+                },
+                "metadata": {
+                    "name": "azure-clusterconfig",
+                    "namespace": "azure-arc",
+                },
+            },
+            "connected_cluster": ConnectedCluster(
+                subscription_id=ZEROED_SUB,
+                cluster_name="cluster1",
+                resource_group_name="rg1",
+            ),
+        },
+        {  # success
+            "failure": False,
+            "config_map": {
+                "apiVersion": "v1",
+                "data": {
+                    "AZURE_RESOURCE_NAME": "cluster1",
+                    "AZURE_RESOURCE_GROUP": "rg1",
+                    "AZURE_SUBSCRIPTION_ID": ZEROED_SUB,
+                },
+                "metadata": {
+                    "name": "azure-clusterconfig",
+                    "namespace": "azure-arc",
+                },
+            },
+            "connected_cluster": ConnectedCluster(
```

Comment:
in all these test cases, the connected_cluster is the same, can you just put it into the test then?